collectors/o365/o365_collector.js

```diff
@@ -168,7 +169,20 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
-            if (err.message) {
+            // set errorCode to showcase client error on DDMetric
+            if (typeof err === 'object') {
+                err.errorCode = err.code ? err.code : (err.statusCode ? err.statusCode : err.status);
+            }
+            // Handle the MS management api maximum palyload issue
+            if (err.message && err.message.indexOf('Maximum payload size exceeded') !== -1) {
+                let min = moment(state.until).diff(state.since, 'minutes');
+                if (min > 1) {
+                    state.until = moment(state.since).add(min / 2, 'minutes').toISOString();
+                }
+                console.warn(`Reduce the time duration by ${min / 2} min due to Maximum payload issue`);
+                return callback(null, [], state, state.poll_interval_sec);
```

Comment:
I think we need to change `state.poll_interval_sec` to `1` second in order to speed up collection